CMSgov/beneficiary-fhir-data
# BFD-2209: Remove database pieces from Synthea automation script

**JIRA Ticket:**
[BFD-2209](https://jira.cms.gov/browse/BFD-2209)

**User Story or Bug Summary:**

As a BFD engineer I would like the synthea automation scripts to not need to contact the database, so that Jenkins (which has no database access) can run them.

---

### What Does This PR Do?


- Removes the pre-validation, post-validation, and characteristics file gen from the run script to facilitate running in Jenkins.
- Moves the non-automation scripts into their own folder, synthea-manual
- Removed validation-related params/code from the synthea run script (which also removed its need for psycopg2 and boto3 and ssmutil)
- Moved pre-validation into its own script (in synthea-manual)
- Add logic to add end state data to the manifest, so that pipeline can validate the load before loading it. This will only be added to the initial load manifest, and not subsequent update manifests. This represents the end state data used to create load, i.e. what should be the current end state data before this data existed, so we can check there are no forseeable conflicts with the data about to be loaded
- Add a small printout to the output of the generation script, to help with the characteristics file generation since it requires start/end bene ids

Manifest additional data will look like this:
![image](https://user-images.githubusercontent.com/1200656/199786268-0e484f5f-68b5-4a2d-b231-14ab18531975.png)


### What Should Reviewers Watch For?


If you're reviewing this PR, please check for these things in particular:

* Verify all PR security questions and checklists have been completed and addressed.


### What Security Implications Does This PR Have?

Submitters should complete the following questionnaire:

* If the answer to any of the questions below is **Yes**, then **you must** supply a link to the associated Security Impact Assessment (SIA), security checklist, or other similar document in Confluence here: **N/A**

    * Does this PR add any new software dependencies? 
    * Does this PR modify or invalidate any of our security controls?
    * Does this PR store or transmit data that was not stored or transmitted before?

* If the answer to any of the questions below is **Yes**, then please add @StewGoin as a reviewer, and note that this PR **should not be merged** unless/until he also approves it.
    * Do you think this PR requires additional review of its security implications for other reasons?

### What Needs to Be Merged and Deployed Before this PR?



This PR cannot be either merged or deployed until the following prerequisite changes have been fully deployed:

* #1431 (BFD-2106) - built off that branch


### Submitter Checklist


I have gone through and verified that...:

* [x] I have named this PR and branch so they are [automatically linked](https://confluence.atlassian.com/adminjiracloud/integrating-with-development-tools-776636216.html) to the (most) relevant Jira issue. Ie: `BFD-123: Adds foo`
    1. It doesn't unnecessarily tie a bunch of disparate features, fixes, refactorings, etc. together.
    2. There isn't too much of a burden on reviewers.
    3. Any problems it causes have a small "blast radius".
    4. It'll be easier to rollback if that becomes necessary.
* [x] The data dictionary has been updated with any field mapping changes, if any were made.
    * At least two other engineers on this project, at least one of whom is a senior engineer or owns the relevant component(s) here.
    * Any relevant engineers on other projects (e.g. DC GEO, BB2, etc.).
    * Please review the standards every few months to ensure you're familiar with them.
